src/api_manager/config.cc

```diff
@@ -569,5 +570,47 @@ std::string Config::GetFirebaseAudience() {
   return GetFirebaseServer() + kFirebaseAudience;
 }
 
+const std::string Config::GetStripedPath(const std::string &path,
+                                         const std::string &base) {
+  if (base.length() == 0) {
+    return path;
+  }
+
+  std::string base_path = base;
+  if (base_path.at(base_path.length() - 1) != '/') {
+    base_path.append("/");
+  }
+
+  std::string stripped = path;
```

Comment:
use string::compare first,  before making a copy.   If doing this in the request,  speed is very important.